### ISSUE TITLE
Contradictory SKI warnings for CABF TLS subscriber certificates

### ISSUE THREAD

**Issue by belghouth** [zmap/zlint#1046]
### Environment

- zlint version: v3.7.0
- Certificate type: CABF TLS DV Subscriber Certificate

Certificate characteristics:

- `basicConstraints = CA:false`
- `EKU = serverAuth`
- SAN present
- AKI present

### Issue

For a TLS subscriber certificate, zlint appears to produce a contradictory/no-win warning condition.

#### Case 1 - SKI omitted

```text
w_ext_subject_key_identifier_missing_sub_cert
```

#### Case 2 - SKI included

```text
w_ext_subject_key_identifier_not_recommended_subscriber
```

### Additional observations

Other linters do not appear to consider the SKI omission a profile issue:

- pkimetal: no findings (even the used zlint inside it)
- x509lint: no findings

```text
cabf.serverauth.subscriber.subject_key_identifier_extension_present
```

### Expected behavior

`w_ext_subject_key_identifier_missing_sub_cert` should not trigger for CABF TLS subscriber certificates with:

- `CA=false`
- subscriber EKU profile
- SAN present
- valid AKI

### Rationale

Modern WebPKI profiles generally discourage Subject Key Identifier usage in TLS subscriber certificates, while SKI remains expected for CA certificates.

As currently implemented, a TLS subscriber certificate cannot avoid one of the two warnings.

Happy to provide reproducer certificates if useful.

**Comment by robstradling** [comment]
In my view, these "contradictory" warnings are expected and correct.  It is a linter's job to provide a report that stays faithful to the requirements document(s) it is linting against.

> Additional observations
> Other linters do not appear to consider the SKI omission a profile issue:
> ...
> pkimetal: no findings (even the used zlint inside it)

When pkimetal is linting according to a TLS BR profile, its zlint handler suppresses `w_ext_subject_key_identifier_missing_sub_cert` (the RFC5280 "SHOULD be present") due to the TLS BR's "NOT RECOMMENDED" - see https://github.com/pkimetal/pkimetal/blob/main/linter/zlint/handler.go#L51

But if you ask pkimetal to lint according to an RFC5280 profile, you should find that it does flag a `w_ext_subject_key_identifier_missing_sub_cert` warning from zlint.

Whilst it's true that the TLS BRs take precedence over RFC5280 in the WebPKI, there are other contexts where RFC5280 is used but the TLS BRs don't apply.

**Comment by belghouth** [comment]
Thank you for clarifying this and for pointing me to the pkimetal filtering logic.

I will tighten my tests and lint selection according to the certificate profile/context being evaluated , 
including using `-includeSources` more explicitly when testing specific profiles/requirements.

Appreciate the clarification.
